baseapp/baseapp.go

```diff
@@ -300,6 +308,72 @@ func (app *BaseApp) QueryRouter() sdk.QueryRouter { return app.queryRouter }
 // GRPCQueryRouter returns the GRPCQueryRouter of a BaseApp.
 func (app *BaseApp) GRPCQueryRouter() *GRPCQueryRouter { return app.grpcQueryRouter }
 
+// RegisterGRPCServer registers gRPC services directly with the gRPC server.
+func (app *BaseApp) RegisterGRPCServer(server gogogrpc.Server) {
+	// Define an interceptor for all gRPC queries: this interceptor will create
+	// a new sdk.Context, and pass it into the query handler.
+	interceptor := func(grpcCtx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		// To create the correct sdk.Context, we first need to transform the
+		// req into an abci.RequestQuery
+		msg, ok := req.(proto.Message)
+		if !ok {
+			return nil, status.Errorf(codes.Internal, "unable to proto marshal")
+		}
+
+		msgBz, err := proto.Marshal(msg)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I actually think this is unnecessary. Instead let's refactor `createQueryContext` to take just `Height` and `Prove` params because those are the only values on `RequestQuery` it actually uses.